tests/exam/test_models.py

```diff
@@ -3,123 +3,103 @@
-    exam_attempt = ExamAttempt.objects.get(user=user, exam=exam)
+    exam_attempt = ExamAttempt.objects.get(id=3)
     self.assertEqual(exam_attempt.calculate_grade(), 0)
 
-    # Check grading for questions with single answer
-    exam_attempt.questions.add(question)
-    answer_attempt = AnswerAttempt.objects.get(attempt=exam_attempt,
-                                               question=question)
-    correct_answer1 = Answer.objects.get(text='Answer 1')
-    answer_attempt.answers.add(correct_answer1)
+  def test_calculate_grade_multiple_choice(self):
+    """Check grading for questions with single answer."""
+
+    self.exam_attempt.questions.add(self.question)
+    self.exam_attempt.grade = 0
+    self.exam_attempt.save()
+
+    self.answer_attempt.answers.set([self.correct_answer1])
+    self.assertEqual(self.exam_attempt.calculate_grade(), 100)
+
+  def test_calculate_grade_multiple_select(self):
+    """Check grading for questions with multiple select answers."""
 
-    self.assertEqual(exam_attempt.calculate_grade(), 100)
+    self.exam_attempt.questions.set([self.question1])
+    self.answer_attempt.question = self.question1
+    self.answer_attempt.save()
+    self.answer_attempt.answers.set(
+        [self.correct_answer1, self.correct_answer2])
 
-    # Now add questions with multiple select answers
-    question1 = Question.objects.get(text='Question 2', exam=exam)
-    exam_attempt.questions.add(question1)
+    self.assertEqual(self.exam_attempt.calculate_grade(), 100)
 
-    answer_attempt2 = AnswerAttempt.objects.create(question=question1,
-                                                   attempt=exam_attempt)
-    correct_answer2 = Answer.objects.get(text='Answer 2')
-    answer_attempt2.answers.add(correct_answer1, correct_answer2)
+  def test_calculate_grade_wrong_answer(self):
+    """Add wrong answer. Should decrease the grade."""
 
-    self.assertEqual(exam_attempt.calculate_grade(), 100)
+    self.exam_attempt.questions.set([self.question1])
+    self.answer_attempt.question = self.question1
+    self.answer_attempt.save()
+    self.answer_attempt.answers.set(
+        [self.correct_answer1, self.correct_answer2])
 
-    # Add wrong answer. Should decrease the grade
-    wrong_answer = Answer.objects.get(text='Answer 3')
-    answer_attempt2.answers.add(wrong_answer)
+    self.answer_attempt.answers.add(self.wrong_answers)
 
-    self.assertEqual(exam_attempt.calculate_grade(), 50)
+    self.assertEqual(self.exam_attempt.calculate_grade(), 0)
 
   def test_passed(self):
     """Check that method correctly checks if exam has been passed."""
 
-    exam = Exam.objects.get(name='Exam 1')
-    exam_attempt = ExamAttempt.objects.get(exam=exam)
-    exam_attempt.grade = 0
-    self.assertEqual(exam_attempt.passed, False)
+    self.exam_attempt.grade = 0
+    self.assertEqual(self.exam_attempt.passed, False)
 
-    exam_attempt.grade = 50
-    self.assertEqual(exam_attempt.passed, False)
+    self.exam_attempt.grade = 50
```

Comment:
You could create a sequence of values for insufficient grade checks. They can be used in a loop with self.assertFalse() afterwards.